File: backend/app/models/movie.py
from datetime import datetime

from pydantic import BaseModel, Field


class Movie(BaseModel):
	id: str | None = None
	item_id: int
	title: str
	genres: list[str]
	overview: str | None = None
	avg_rating: float | None = 0.0
	rating_number: int | None = 0
	sentiment_label: str | None = None
	created_at: datetime = Field(default_factory=datetime.utcnow)

File: frontend/src/types/movie.ts
export interface Movie {
  id: number;
  item_id?: number;
  title: string;
  genres: string[];
  overview?: string | null;
  avg_rating?: number | null;
  rating_number?: number | null;
  created_at?: string;
  sentiment_label?: string | null;
}
